src/hmem_cuda.c

```diff
@@ -754,12 +725,33 @@ int cuda_host_unregister(void *ptr)
 
 enum cuda_xfer_setting cuda_get_xfer_setting(void)
 {
+	enum cuda_xfer_setting hmem_cuda_xfer_setting;
+	int ret, param_value = 1;
+
+	/* Fetch the value of the environment variable FI_HMEM_CUDA_ENABLE_XFER */
+	ret = fi_param_get_bool(NULL, "hmem_cuda_enable_xfer", &param_value);
+	if (ret == -EINVAL){
+		fprintf(stderr, "FI_HMEM_CUDA_ENABLE_XFER was set to an invalid value by the user."
+			" FI_HMEM_CUDA_ENABLE_XFER is boolean and can be set to only 0/false/no/off or"
+			" 1/true/yes/on.\n");
+		abort();
```

Comment:
Aborting the application is too harsh of a response to an invalid environment variable.  Just use the default.